src/packs/nzism-v36-1022-20.ts

```diff
@@ -0,0 +1,845 @@
+      node: node,
+    });
+  }
+
+  /**
+   * Check CloudFront Resources
+   * @param node
+   * @param ignores
+   */
+  private checkCloudFront(node: CfnResource): void {
+    this.applyRule({
+      info: 'SHOULD 16.6.10.C.02[CID:2013], MUST 23.5.11.C.01[CID:7496]| Access Control and Passwords/Event Logging and Auditing/Additional events to be logged & Public Cloud Security/Logging and Alerting in Public Cloud/Logging requirements',
+      explanation: 'CloudFrontDistributionAccessLogging',
+      level: NagMessageLevel.ERROR,
+      rule: CloudFrontDistributionAccessLogging,
+      node: node,
+    });
+    this.applyRule({
+      info: 'MUST 19.1.12.C.01[CID:3562]| Gateway security/Gateways/Configuration of gateways',
+      explanation:
+        'The Web Application Firewall can help protect against application-layer attacks that can compromise the security of the system or place unnecessary load on them.',
+      level: NagMessageLevel.ERROR,
+      rule: CloudFrontDistributionWAFIntegration,
+      node: node,
+    });
+
+    this.applyRule({
+      info: 'MUST 16.1.37.C.01[CID:1847]| Access Control and Passwords/Identification, Authentication and Passwords/Protecting authentication data in transit',
+      explanation:
+        'Vulnerabilities have been and continue to be discovered in the deprecated SSL and TLS protocols. Help protect viewer connections by specifying a viewer certificate that enforces a minimum of TLSv1.1 or TLSv1.2 in the security policy. Distributions that use the default CloudFront viewer certificate or use vip for the SslSupportMethod are non-compliant with this rule, as the minimum security policy is set to TLSv1 regardless of the specified MinimumProtocolVersion',
+      level: NagMessageLevel.ERROR,
+      rule: CloudFrontDistributionHttpsViewerNoOutdatedSSL,
+      node: node,
+    });
+  }
+
+  /**
+   * Check CloudTrail Resources
+   * @param node the CfnResource to check
+   * @param ignores list of ignores for the resource
+   */
+  private checkCloudTrail(node: CfnResource): void {
+    this.applyRule({
+      info: 'SHOULD 16.6.6.C.02[CID:1998], MUST 16.4.35.C.02[CID:6860]| Access Control and Passwords: (Event Logging and Auditing/Maintaining system management logs & Privileged Access Management/Monitoring and Review)',
+      explanation:
+        'Use Amazon CloudWatch to centrally collect and manage log event activity. Inclusion of AWS CloudTrail data provides details of API call activity within your AWS account.',
+      level: NagMessageLevel.ERROR,
+      rule: CloudTrailCloudWatchLogsEnabled,
+      node: node,
+    });
+
+    this.applyRule({
+      info: 'SHOULD 17.1.46.C.04[CID:2082]| Cryptography/Cryptographic Fundamentals/Reducing storage and physical transfer requirements',
+      explanation:
+        'Because sensitive data may exist and to help protect data at rest, ensure encryption is enabled for your AWS CloudTrail trails.',
+      level: NagMessageLevel.ERROR,
```

Comment:
General comment here not knowing much about these rules. Is it wrong to assume that anything with only  a `SHOULD` control ID be marked as a `Warning` instead of an `Error`?

Reply:
That is a good point.       'SHOULD' should map to `Warning`  and MUST to  `Error`.   Where the rule maps to multiple controls,  we should use the 'strongest' level, if there are both MUST and SHOULD..          Have modified this.